src/ucp/core/ucp_worker.inl

```diff
@@ -96,7 +96,8 @@ ucp_worker_del_request_id(ucp_worker_h worker, ucp_request_t *request,
 {
     ucs_status_t status UCS_V_UNUSED;
 
-    ucs_assert(request->flags & UCP_REQUEST_FLAG_IN_PTR_MAP);
+    ucs_assert((request->flags & UCP_REQUEST_FLAG_IN_PTR_MAP) ||
+               !ucs_ptr_map_key_indirect(id));
```

Comment:
maybe add helper for this assert? seems to be in many places

Reply:
makes sense, but need two different helper functions